gallifrey/backend/src/main/java/gallifrey/backend/VectorClockBackend.java

```diff
@@ -186,15 +190,46 @@ public OtpErlangTuple snapshot(OtpErlangBinary JavaObjectId) throws NoSuchObject
             // object
             TreeSet<GenericEffect> crdt_effect_buffer = mapentry.effectbuffer;
             TreeSet<GenericEffect> new_crdt_effect_buffer = new TreeSet<GenericEffect>();
-            for (GenericEffect e : crdt_effect_buffer) {
-                // Use a correct compare based on types
-                if (e.time.lessthan(this.GlobalClockTime) || (0 == e.time.compareTo(this.GlobalClockTime))
-                        || this.GlobalClockTime.isEmpty()) {
+            ArrayList<ArrayList<GenericEffect>> grouped_by_merge_strategy = new ArrayList<>();
+            {
+                ArrayList<GenericEffect> current_group = null;
+                for (GenericEffect e : crdt_effect_buffer) {
+                    if (e.time.lessthan(this.GlobalClockTime) || (0 == e.time.compareTo(this.GlobalClockTime))
+                            || this.GlobalClockTime.isEmpty()) {
+                        MergeComparator merge_strategy = e.get_merge_strategy();
+                        if (current_group == null || current_group.get(0).get_merge_strategy().equals(merge_strategy)) {
+                            grouped_by_merge_strategy.add(new ArrayList<GenericEffect>());
+                            current_group = grouped_by_merge_strategy.get(grouped_by_merge_strategy.size() - 1);
```

Comment:
Is this a round about way of doing:
`current_group = new ArrayList<GenericEffect>();
grouped_by_merger_strategy.add(current_group);`